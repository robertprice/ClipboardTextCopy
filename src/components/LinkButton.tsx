import React from 'react';

import { Button } from './Button';
import { Link } from './Link';

export interface LinkButtonProps {
  labelCaption: string;
  tooltip: string;
  iconUrl: string;
  renderType: string;
  buttonType?: ButtonType;
  buttonClass?: string;
  tabIndex: number;
  onClick: (event: React.MouseEvent<HTMLElement>) => void;
}

export const LinkButton = (props: LinkButtonProps) => {
  if (props.renderType === 'button')
    return (
      <Button
        label={props.labelCaption}
        onClickAction={props.onClick}
        iconUrl={props.iconUrl}
        tooltip={props.tooltip}
        className={props.buttonClass}
        tabIndex={props.tabIndex}
        buttonType={props.buttonType}
      />
    );
  if (props.renderType === 'link')
    return (
      <Link
        label={props.labelCaption}
        onClickAction={props.onClick}
        iconUrl={props.iconUrl}
        tooltip={props.tooltip}
        tabIndex={props.tabIndex}
      />
    );
  return null;
};
