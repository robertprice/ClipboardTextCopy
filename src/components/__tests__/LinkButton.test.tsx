import React from 'react';
import { create } from 'react-test-renderer';

import parseStyle from '../../utils/parseStyle';
import { LinkButton } from '../LinkButton';

const handleClick = (event: React.MouseEvent<HTMLElement>) => {
  event.preventDefault();
  event.stopPropagation();
  event.nativeEvent.stopImmediatePropagation();
  return false;
};

const sampleStyle = parseStyle(
  'border-style:solid;border-color:#39c;border-width:5px;'
);

describe('ClipboardTextCopy Component Test Unit', () => {
  it('should render button type correctly', () => {
    const props = {
      onClick: handleClick,
      className: 'awsomeClass',
      tabIndex: 2,
      style: sampleStyle,
      labelCaption: 'labelCaption',
      tooltip: 'buttonTitle',
      renderType: 'button',
      iconUrl: 'icon',
      buttonType: 'info',
      buttonClass: 'jumbotron',
    };
    const component = create(<LinkButton {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render link type correctly', () => {
    const props = {
      onClick: handleClick,
      className: 'awsomeClass',
      tabIndex: 2,
      style: sampleStyle,
      labelCaption: 'labelCaption',
      tooltip: 'linkTitle',
      renderType: 'link',
      iconUrl: 'icon',
    };
    const component = create(<LinkButton {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
