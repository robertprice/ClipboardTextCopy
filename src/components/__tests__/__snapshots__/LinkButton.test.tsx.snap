// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClipboardTextCopy Component Test Unit should render button type correctly 1`] = `
<button
  className="clipboardTextCopy btn mx-button jumbotron btn-info"
  onClick={[Function]}
  tabIndex={2}
  title="buttonTitle"
  type="button"
>
  <img
    className="clipboardTextCopy-img"
    src="icon"
  />
  labelCaption
</button>
`;

exports[`ClipboardTextCopy Component Test Unit should render link type correctly 1`] = `
<span
  className="clipboardTextCopy mx-link"
  title="linkTitle"
>
  <img
    className="clipboardTextCopy-img"
    src="icon"
  />
  <a
    href=""
    onClick={[Function]}
    tabIndex={2}
  >
    labelCaption
  </a>
</span>
`;
